shared/src/main/scala/pl/metastack/metatime/PatternTime.scala

```diff
@@ -0,0 +1,66 @@
+package pl.metastack.metatime
+
+case class PatternTime(parts: Seq[Time => String] = Nil) {
+  def string(s: String): PatternTime = add(_ => s)
+
+  def dash : PatternTime = string("-")
```

Comment:
This is duplicated in `PatternDateTime`. Can you create a trait `Pattern` that contains shared functionality?